### ISSUE TITLE
Determine the zone from the domain

### ISSUE THREAD

**Issue by ttocsr** [aaomidi/certbot-dns-google-domains#14]
This is a enhancement request.
Most other Authenticators usually have this added to the credentials file.
There is no means of adding --dns-google-domains-zone to nginx proxy manager for example.
https://github.com/NginxProxyManager/nginx-proxy-manager/blob/develop/global/certbot-dns-plugins.js

**Comment by ttocsr** [comment]
Or alternatively could this not be determined automatically without the use of an additional flag by parsing -d input?
If -d www.example.com pass the zone as example.com

**Comment by DeveloperTommy** [comment]
On the alternative, parsing the -d input may be error prone because there are domains with TLDs such as .co.uk that can break this logic depending on how the parsing is implemented. If there's already another Python library that does this, that might be worthwhile so we can reduce maintenance burden for new TLDs.

**Comment by ttocsr** [comment]
Certbot has tldextract in tools/requirements.txt. But I don't see it actually used anywhere.
It should handle those specific cases.  I'm just reading through how zones are implemented automatically other places with certbot.

**Comment by aaomidi** [comment]
Interesting, I can certainly add both options here. I thought credentials might be a bit annoying in case someone is maintaining multiple domains there.

**Comment by ttocsr** [comment]
I agree.  When I opened the issue I posted about using credentials because it just happened to be the first example of a zone I had found.  Turns out that was the only example of using credentials.    Looks like the common way is a function to get the zone from the API. But the documentation for the API doesn't have anything documented for that.
Personally I think we're left with figuring the TLD from the -d domain
